packages/dashboard/src/dashboard-actions/createWidget.ts

```diff
@@ -0,0 +1,33 @@
+import { DashboardConfiguration, Widget } from '../types';
+import { concatWidgets } from '../util/dashboardConfiguration';
+import { MockWidgetFactory } from '../testing/mocks';
+
+export const createWidget = ({
+  dashboardConfiguration,
+  widgets,
+}: {
+  dashboardConfiguration: DashboardConfiguration;
+  widgets: Widget[];
+}): DashboardConfiguration => {
+  dashboardConfiguration = concatWidgets(dashboardConfiguration, widgets);
+  return dashboardConfiguration;
+};
+
+const getRandomInt = (min: number, max: number) => {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+};
+
+export const getRandomWidget = () => {
```

Comment:
Can we move this into the testing folder?

Reply:
we could! however, it is one of the functions that the reducer calls and is undo/redoable so it feels a bit more appropriate in dashboard-actions to me

Reply:
I think the getRandom functions would only be used in testing though?

Reply:
good point, that is only going to be used for testing but i dont imagine the createWidget only being used for testing, so i'm a little conflicted. maybe I should just take out the randomization for now?